File: Java/bitManipulation.java
public class bitManipulation {
    public static int binaryToDecimal (int N) {
        int dec = 0;
        int power = 1;
        while( N!=0 ){
            int r = N%10;
            dec += power*r;
            power *= 2;
            N = N/10;
        }
        return dec;
    }

    public static int decimalToBinary (int N) {
        int bin = 0;
        int power = 1;
        while( N!=0 ){
            int r = N%2;
            bin += power*r;
            power *= 10;
            N = N/2;
        }
        return bin;
    }

    public static void main(String args[]) {
        int A = 1011;
        int B = 11;
        System.out.println(binaryToDecimal(A));
        System.out.println(decimalToBinary(B));
    }
}

File: JS/bitManipulation.js
/*------------ Binary to Decimal -----------------*/
function binaryToDecimal(n) {
  let dec = 0;
  let p = 1; // power factor
  while (n != 0) {
    // get last digit
    let r = n % 10;
    // add contribution
    dec += r * p;
    // next power of 2
    p *= 2;
    // remove last digit
    n = Math.floor(n / 10);
  }
  return dec;
}

/*------------ Decimal to Binary -----------------*/
function decimalToBinary(n) {
  let bin = 0;
  let p = 1;
  while (n != 0) {
    let r = n % 2;
    bin += r * p;
    p *= 10;
    n = Math.floor(n / 2);
  }
  return bin;
}

console.log(binaryToDecimal(1011));
console.log(decimalToBinary(11));
